providers/azure.go

```diff
@@ -81,16 +81,12 @@ func NewAzureProvider(p *ProviderData, opts options.AzureOptions) *AzureProvider
 	}
 	p.getAuthorizationHeaderFunc = makeAzureHeader
 
-	tenant := "common"
```

Comment:
I am not sure if I read you correctly: I believe there are no other places where defaults for the Azure provider are set?

Reply:
Ahh right so you moved the default out of here, hmm, does the override logic work when the tenant is `common`? Wondering if we need a check that blocks the override when the tenant is set to `common`